src/elements/content-explorer/RenameDialog.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { Modal, TextInput } from '@box/blueprint-web';
+import type { BoxItem } from '../../common/types/core';
+
+import { ERROR_CODE_ITEM_NAME_TOO_LONG, ERROR_CODE_ITEM_NAME_IN_USE } from '../../constants';
+
+import messages from '../common/messages';
+
+import './RenameDialog.scss';
+
+export interface RenameDialogProps {
+    errorCode: string;
+    isLoading: boolean;
+    isOpen: boolean;
+    item: BoxItem;
+    onCancel: () => void;
+    onRename: (nameWithoutExt: string, extension: string) => void;
+    parentElement: HTMLElement;
+}
+
+/* eslint-disable jsx-a11y/label-has-for */
+const RenameDialog = ({ errorCode, isLoading, isOpen, item, onCancel, onRename, parentElement }: RenameDialogProps) => {
+    const { formatMessage } = useIntl();
+
+    let textInput = null;
+    let error;
+
+    const { name = '', extension, type } = item;
+    const ext = extension ? `.${extension}` : '';
+    const nameWithoutExt = extension ? name.replace(ext, '') : name;
+    const itemType = type === 'web_link' ? 'webLink' : type;
+
+    /**
+     * Appends the extension and calls rename function
+     */
+    const rename = () => {
+        if (textInput && textInput.value) {
+            if (textInput.value === nameWithoutExt) {
+                onCancel();
+            } else {
+                onRename(textInput.value, ext);
+            }
+        }
+    };
+
+    /**
+     * Grabs reference to the input element
+     */
+    const ref = input => {
+        textInput = input;
+        if (textInput instanceof HTMLInputElement) {
+            textInput.focus();
+            textInput.select();
+        }
+    };
+
+    /**
+     * Handles enter key down
+     */
+    const onKeyDown = ({ key }) => {
+        switch (key) {
+            case 'Enter':
+                rename();
+                break;
+            default:
+                break;
+        }
+    };
+
+    switch (errorCode) {
+        case ERROR_CODE_ITEM_NAME_IN_USE:
+            error = messages.renameDialogErrorInUse;
+            break;
+        case ERROR_CODE_ITEM_NAME_TOO_LONG:
+            error = messages.renameDialogErrorTooLong;
+            break;
+        default:
+            error = errorCode ? messages.renameDialogErrorInvalid : null;
+            break;
+    }
+
+    return (
+        <Modal onOpenChange={onCancel} open={isOpen}>
```

Comment:
i don't think `onOpenChange` is used like this? the callback should be for both opening and closing

Reply:
talked on zoom. we're going to just leave this as-is